components/omnibox/browser/leo_provider_unittest.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/omnibox/browser/leo_provider.h"
+
+#include "base/memory/scoped_refptr.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/omnibox/browser/brave_fake_autocomplete_provider_client.h"
+#include "components/omnibox/browser/autocomplete_input.h"
+#include "components/omnibox/browser/test_scheme_classifier.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace {
+
+class LeoEnabledAutocompleteProviderClient
+    : public BraveFakeAutocompleteProviderClient {
+ public:
+  LeoEnabledAutocompleteProviderClient() = default;
+  ~LeoEnabledAutocompleteProviderClient() override = default;
+  bool IsLeoProviderEnabled() override { return true; }
+};
+
+class LeoProviderTest : public testing::Test {
```

Comment:
same here. tests should not be in anonymous namespace. You can't `friend` a class from anonymous ns.

Reply:
Fixed in 8044f1b72e2e62c3bdd8cd34c66e39fec50bb8d2